src/Mapbender/ManagerBundle/Resources/public/js/application-edit.js

```diff
@@ -324,6 +319,31 @@ $(function() {
         return false;
     });
 
+    function elementFormSubmit() {
+        var form = $("#elementForm"),
+            data = form.serialize(),
+            url = form.attr('action'),
+            self = this;
+
+        $.ajax({
+            url: url,
+            method: 'POST',
+            data: data,
+            error: function (e, statusCode, message) {
+                alert(Mapbender.trans("mb.application.save.failure.general") + ' ' + message);
+            },
+            success: function(data) {
+                if (data.length > 0) {
+                    $(form.parent()).html( data );
+                } else {
+                    $("#elementForm").data('dirty', false);
+                    self.close();
+                    window.location.reload();
```

Comment:
Why do you reload the page here? Is it really necessary? When it is necessary, is there an advantage in closing the form before?

Reply:
Because the usual server response to a successful POST (or PUT; we don't support form methods, apparently!) is to redirect to the modified resource view with a 303.

We kind of "emulate" the traditional behavior here.